aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -88,6 +90,19 @@ void pythonFallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
           return;
         }
       }
+    } else if (ivalue.isSymIntList()) {
+      for (const auto& nv : ivalue.toListRef()) {
+        if (nv.isNone()) {
+          continue;
+        }
+        auto x = nv.toSymInt();
+        if (x.is_heap_allocated() && x.toSymNode()->is_singleton()) {
+          const auto& nt_cls = at::impl::get_nested_tensor_cls();
+          TORCH_INTERNAL_ASSERT(nt_cls);
+          nt_cls->pyinterpreter()->dispatch(op, stack);
+          return;
```

Comment:
I'd be happier if this wasn't so NT specific. For example, can't we just have a single well known pyinterpreter that is registered at interpreter startup and dispatch to that? Then the only test is "should this SymInt be dispatched to Python", which could be a dedicated method on SymNode

Reply:
Updated to have a single pyinterpreter registered when we init the modules, and to no longer register the class at module import time. Instead, just look up the class's pyobject when we dispatch (and cache the result).